datafusion/sql/src/unparser/expr.rs

```diff
@@ -174,133 +225,243 @@ impl Unparser<'_> {
             }
-            ScalarValue::Utf8(None) => Ok(ast::Value::Null),
-            ScalarValue::LargeUtf8(Some(str)) => {
-                Ok(ast::Value::SingleQuotedString(str.to_string()))
+            ScalarValue::Int64(None) => Ok(ast::Expr::Value(ast::Value::Null)),
+            ScalarValue::UInt8(Some(ui)) => {
+                Ok(ast::Expr::Value(ast::Value::Number(ui.to_string(), false)))
             }
-            ScalarValue::LargeUtf8(None) => Ok(ast::Value::Null),
+            ScalarValue::UInt8(None) => Ok(ast::Expr::Value(ast::Value::Null)),
+            ScalarValue::UInt16(Some(ui)) => {
+                Ok(ast::Expr::Value(ast::Value::Number(ui.to_string(), false)))
+            }
+            ScalarValue::UInt16(None) => Ok(ast::Expr::Value(ast::Value::Null)),
+            ScalarValue::UInt32(Some(ui)) => {
+                Ok(ast::Expr::Value(ast::Value::Number(ui.to_string(), false)))
+            }
+            ScalarValue::UInt32(None) => Ok(ast::Expr::Value(ast::Value::Null)),
+            ScalarValue::UInt64(Some(ui)) => {
+                Ok(ast::Expr::Value(ast::Value::Number(ui.to_string(), false)))
+            }
+            ScalarValue::UInt64(None) => Ok(ast::Expr::Value(ast::Value::Null)),
+            ScalarValue::Utf8(Some(str)) => Ok(ast::Expr::Value(
+                ast::Value::SingleQuotedString(str.to_string()),
+            )),
+            ScalarValue::Utf8(None) => Ok(ast::Expr::Value(ast::Value::Null)),
+            ScalarValue::LargeUtf8(Some(str)) => Ok(ast::Expr::Value(
+                ast::Value::SingleQuotedString(str.to_string()),
+            )),
+            ScalarValue::LargeUtf8(None) => Ok(ast::Expr::Value(ast::Value::Null)),
             ScalarValue::Binary(Some(_)) => not_impl_err!("Unsupported scalar: {v:?}"),
-            ScalarValue::Binary(None) => Ok(ast::Value::Null),
+            ScalarValue::Binary(None) => Ok(ast::Expr::Value(ast::Value::Null)),
             ScalarValue::FixedSizeBinary(..) => {
                 not_impl_err!("Unsupported scalar: {v:?}")
             }
             ScalarValue::LargeBinary(Some(_)) => {
                 not_impl_err!("Unsupported scalar: {v:?}")
             }
-            ScalarValue::LargeBinary(None) => Ok(ast::Value::Null),
+            ScalarValue::LargeBinary(None) => Ok(ast::Expr::Value(ast::Value::Null)),
             ScalarValue::FixedSizeList(_a) => not_impl_err!("Unsupported scalar: {v:?}"),
             ScalarValue::List(_a) => not_impl_err!("Unsupported scalar: {v:?}"),
             ScalarValue::LargeList(_a) => not_impl_err!("Unsupported scalar: {v:?}"),
-            ScalarValue::Date32(Some(_d)) => not_impl_err!("Unsupported scalar: {v:?}"),
-            ScalarValue::Date32(None) => Ok(ast::Value::Null),
-            ScalarValue::Date64(Some(_d)) => not_impl_err!("Unsupported scalar: {v:?}"),
-            ScalarValue::Date64(None) => Ok(ast::Value::Null),
+            ScalarValue::Date32(Some(d)) => {
```

Comment:
I wonder if there is some function we can use in arrow-rs do to this conversion

Perhaps https://docs.rs/arrow/latest/arrow/array/types/struct.Date32Type.html#method.to_naive_date ?

Reply:
Good call. I was able to remove the chrono explicit dependency and downcast to arrow arrays and use those methods instead.